src/MLJGLMInterface.jl

```diff
@@ -168,19 +190,42 @@ glm formula and glm data header.
 """
 function glm_features(model, X)
     if Tables.columnaccess(X)
-        table_features = keys(Tables.columns(X))
+        table_features = _to_array(keys(Tables.columns(X)))
     else
         first_row = iterate(Tables.rows(X), 1)[1]
-        table_features = first_row === nothing ? Symbol[] : keys(first_row)
+        table_features = first_row === nothing ? Symbol[] : _to_array(keys(first_row))
     end
-    features = filter(x -> x != model.offsetcol, table_features)
+    features = filter!(x -> x != model.offsetcol, table_features)
     return features
 end
 
 ####
 #### FIT FUNCTIONS
 ####
 
+struct FitResult{V<:AbstractVector, T, R}
+    "Vector containg coeficients of the predictors and intercept"
+    coefs::V
+    "An estimate of the dispersion parameter of the glm model. "
+    dispersion::T
+    "Other fitted parameters specific to a fitted model"
+    params::R
+    function FitResult(
+        coefs::AbstractVector,
+        dispersion,
+        params
+    )
+        return new{
+            typeof(coefs),
+            typeof(dispersion),
+            typeof(params)
+        }(coefs, dispersion, params)
```

Comment:
```suggestion
```
It works also without this inner constructor right? Julia creates one automatically

Reply:
Yeah, I know it does. I was just trying to get rid of some annoying code highlighting vscode was showing me.